yarn-project/p2p/src/service/libp2p_service.ts

```diff
@@ -380,7 +411,39 @@ export class LibP2PService implements P2PService {
     const txHash = tx.getTxHash();
     const txHashString = txHash.toString();
     this.logger.verbose(`Received tx ${txHashString} from external peer.`);
-    await this.txPool.addTxs([tx]);
+
+    if (await this.isValidTx(tx)) {
+      await this.txPool.addTxs([tx]);
+    }
+  }
+
+  // TODO: these should be done separately and results will be used for peer scoring
+  private async isValidTx(tx: Tx): Promise<boolean> {
+    const blockNumber = (await this.l2BlockSource.getBlockNumber()) + 1;
+
+    const newGlobalVariables = await this.globalVariableBuilder.buildGlobalVariables(
```

Comment:
Oof, this is an expensive call to be doing on every transaction. It makes lots of calls to the contract. We already have the block number and we just need the chain id, which never changes. Can you update the MetadataTxValidator to just take the data it needs and get rid of this global variable stuff?